src/verify/index.ts

```diff
@@ -0,0 +1,52 @@
+import { isEvmAddress, isStarknetAddress } from '../utils';
+import verifyStarknetMessage, {
+  type NetworkType,
+  getHash as getStarknetHash
+} from './starknet';
+import verifyEvmMessage, { getHash as getEvmHash } from './evm';
+import type { ProviderOptions } from '../utils/provider';
+import type { StarkNetType } from 'starknet';
+import type { TypedDataField } from '@ethersproject/abstract-signer';
+
+export type SignaturePayload = {
+  domain: Record<string, string>;
+  types: Record<string, StarkNetType[] | TypedDataField[]>;
+  primaryType?: string;
+  message: Record<string, any>;
+};
+
+export function getHash(data: SignaturePayload, address?: string): string {
+  if (data.primaryType && address) {
+    return getStarknetHash(data, address);
+  }
+
+  return getEvmHash(data);
+}
```

Comment:
I didn't try it, but I think It is possible that someone may pass primaryType in their payload for EVM signature 🤔 on ingestor we are doing this
`const id = snapshot.utils.getHash(body.data, body.address);` so address will be always passed, 

so I have feeling that we may always get a starknet hash, I will try to test tmrw

Reply:
We can add a check on the presence of `StarknetDomain` in the types.

Reply:
Maybe we can do same as below? based on `address` if it is startkent address, we could get starknet hash and if it is a evm address or if no address is passed, we can get EVM hash

Reply:
The starknet address detection is flimsy, as all evm address are valid starknet address

Better to analyze the payload in that case

Reply:
Refactored to analyze message payload instead of address to detect starknet/evm message